### DISCUSSION TITLE
information about training

### DISCUSSION THREAD

**Discussion by Sarath-C-Koppolu** [ultralytics/ultralytics#13656] [General]
from ultralytics import YOLO

# Load a pretrained model
model = YOLO("yolov8n.pt")  # Ensure this is the correct path to the model config file

# Specify the dataset configuration file for fold 1
dataset_yaml = r'C:\Users\sarat\Desktop\Final\sample.v1i.yolov9 (1)\CV_Final\fold_1\data.yaml'

# Train the model with all arguments included in the train method call
model.train(
    data=dataset_yaml,
    epochs=250,
    patience=25,
    batch=4,
    project='V8_N_480',
    name='N_480_IMAGE',
    imgsz=480,
    lr0=0.01,
    optimizer='auto',
    device=0,
    cos_lr=True,
    workers=10,
)

# Save output metrics for further analysis

when i use imgz as 320 the code is running super fine 
and when i run with 480 the mAP are starting from 0.02 and continoue to decrease. then i tried to check it with 640, 800, 960 still the same infact with 800 and 960 it is strating with 0 as mAP and it is increasing

i tried to change learning rate from 0.01 to 0.001 as well but still it is the same 

and i used v9C model with the same code and i have seen the similar results, mAP starts from 0.02 and it is decreasing with 2nd epoch 0.0003

i have no clue why it is happening.. 

please respond

**Comment by glenn-jocher** [comment]
Hi @Sarath-C-Koppolu,

Thank you for providing the details of your training setup. Let's work through this together to identify the issue.

1. **Reproducible Code Example**: Your code snippet is helpful, but to further investigate, could you please provide a minimum reproducible example? This will help us replicate the issue on our end. You can refer to our guide on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

2. **Version Check**: Ensure you are using the latest versions of `torch` and `ultralytics`. You can upgrade them using the following commands:
    ```bash
    pip install --upgrade torch ultralytics
    ```

3. **Image Size Impact**: The issue you're experiencing with different image sizes might be related to how the model handles varying resolutions. Here are a few steps to troubleshoot:
    - **Learning Rate**: You've already tried adjusting the learning rate, which is good. Sometimes, larger image sizes require a different learning rate schedule.
    - **Batch Size**: Larger images consume more memory, which might affect the batch size. Ensure your GPU has enough memory to handle the larger images with the specified batch size.
    - **Model Configuration**: Verify that the model configuration is appropriate for the image sizes you're using. Sometimes, certain architectures might need adjustments when changing image sizes.

4. **Training Metrics**: Monitor other metrics such as loss, precision, and recall during training to get a better understanding of what's happening. Tools like TensorBoard can be very useful for this.

5. **Example Adjustments**: Here’s a slightly adjusted version of your training script to ensure all configurations are set correctly:
    ```python
    from ultralytics import YOLO

    # Load a pretrained model
    model = YOLO("yolov8n.pt")

    # Specify the dataset configuration file for fold 1
    dataset_yaml = r'C:\Users\sarat\Desktop\Final\sample.v1i.yolov9 (1)\CV_Final\fold_1\data.yaml'

    # Train the model with all arguments included in the train method call
    model.train(
        data=dataset_yaml,
        epochs=250,
        patience=25,
        batch=4,
        project='V8_N_480',
        name='N_480_IMAGE',
        imgsz=480,
        lr0=0.01,
        optimizer='auto',
        device=0,
        cos_lr=True,
        workers=10,
    )
    ```

If the issue persists, please share any error messages or logs that might provide more insight. Additionally, you can visit our [YOLO Common Issues Guide](https://docs.ultralytics.com/guides/yolo-common-issues/) for more troubleshooting tips.

Looking forward to your response!
